test/ntp_client_test.rb

```diff
@@ -0,0 +1,487 @@
+        subject.Write
+      end
+
+      context "when it is in write only mode" do
+        it "doesn't try to restart ntp service" do
+          subject.write_only = true
+          expect(Yast::Service).not_to receive(:Restart).with(subject.service_name)
+
+          subject.Write
+        end
+      end
+
+      context "when it is not in write only mode" do
+        it "tries to restart ntp service" do
+          expect(Yast::Service).to receive(:Restart).with(subject.service_name)
+
+          subject.Write
+        end
+      end
+    end
+
+    context "when has not been configured to run as a service" do
+      before do
+        subject.run_service = false
+      end
+
+      it "disables ntp service" do
+        expect(Yast::Service).to receive(:Disable).with(subject.service_name)
+        expect(Yast::Service).to receive(:Stop).with(subject.service_name)
+        subject.Write
+      end
+    end
+
+    context "when synchronize time is false" do
+      it "writes cronfile to sync ntp via cron" do
+        subject.synchronize_time = false
+        expect(Yast::SCR).to receive(:Execute)
+          .with(Yast::Path.new(".target.bash"),
+            "test -e #{subject.cron_file} && rm #{subject.cron_file};"
+               )
+
+        subject.Write
+      end
+    end
+
+    context "when synchronize time is true" do
+      let(:cron_entry) do
+        "-*/#{subject.sync_interval} * * * * root /usr/sbin/start-ntpd ntptimeset &>/dev/null\n"
+      end
+      it "writes cronfile to sync ntp via cron" do
+        subject.synchronize_time = true
+        expect(Yast::SCR).to receive(:Write)
+          .with(Yast::Path.new(".target.string"), subject.cron_file, cron_entry)
+
+        subject.Write
+      end
+    end
+
+    it "returns true if not aborted" do
+      expect(subject.Write).to eql(true)
+    end
+  end
+
+  describe "#GetNtpServers" do
+    let(:country_servers) do
+      { "tick.nap.com.ar" =>
+        { "access_policy "  => "open access, please send a message to notify",
+          "address"         => "tick.nap.com.ar",
+          "country"         => "AR",
+          "exact_location"  => "Network Access Point, Buenos Aires, Argentina",
+          "location"        => "Argentina",
+          "stratum"         => "2",
+          "synchronization" => "NTP V3 secondary (stratum 2), Cisco IOS"
+        }
+      }
+    end
+
+    context "when ntp servers haven't been read before" do
+      it "returns empty hash in test Mode" do
+        allow(Yast::Mode).to receive(:test).and_return(true)
+
+        expect(subject.GetNtpServers).to eql({})
+      end
+
+      it "caches known ntp servers" do
+        allow(Yast::Mode).to receive(:test).and_return(false)
+        subject.instance_variable_set(:@ntp_servers, nil)
+        expect(subject).to receive(:update_ntp_servers!)
+
+        subject.GetNtpServers
+      end
+
+      it "return known ntp servers" do
+        subject.instance_variable_set(:@ntp_servers, country_servers)
```

Comment:
You shouldn't rely on instance variables. You're loading NTP servers from a YAML file, right? So you can check that those servers are returned.